graphql_compiler/tests/schema_transformation_tests/test_rename_schema.py

```diff
@@ -289,6 +289,175 @@ def test_cyclic_rename(self) -> None:
         )
         self.assertEqual({}, renamed_schema.reverse_field_name_map)
 
+    def test_field_rename(self) -> None:
+        renamed_schema = rename_schema(parse(ISS.basic_schema), {}, {"Human": {"id": {"new_id"}}})
+        renamed_schema_string = dedent(
+            """\
+            schema {
+              query: SchemaQuery
+            }
+
+            directive @stitch(source_field: String!, sink_field: String!) on FIELD_DEFINITION
+
+            type Human {
+              new_id: String
+            }
+
+            type SchemaQuery {
+              Human: Human
+            }
+        """
+        )
+        compare_schema_texts_order_independently(
+            self, renamed_schema_string, print_ast(renamed_schema.schema_ast)
+        )
+        self.assertEqual({}, renamed_schema.reverse_name_map)
+        self.assertEqual({"Human": {"new_id": "id"}}, renamed_schema.reverse_field_name_map)
+
+    def test_field_rename_includes_original_name(self) -> None:
+        renamed_schema = rename_schema(
+            parse(ISS.basic_schema), {}, {"Human": {"id": {"new_id", "id"}}}
+        )
+        renamed_schema_string = dedent(
+            """\
+            schema {
+              query: SchemaQuery
+            }
+
+            directive @stitch(source_field: String!, sink_field: String!) on FIELD_DEFINITION
+
+            type Human {
+              id: String
+              new_id: String
```

Comment:
Shouldn't this lead to a name conflict error because there are two fields named "new_id"?

Reply:
Nope, there's just one `new_id`-- this is a separate test from the previous test, `test_field_rename`

Reply:
Yep you are right.